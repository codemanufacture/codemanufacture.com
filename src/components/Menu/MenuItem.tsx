import * as React from 'react'
import { Link } from 'gatsby'
import MenuItemInterface from '../Menu/MenuItemInterface'

interface MenuItemProps {
  item: MenuItemInterface
}

const MenuItem: React.SFC<MenuItemProps> = props => (
  <li>
    <Link to={`/homepage#${props.item.sectionId}`} title={props.item.name}>
      {props.item.name}
    </Link>
  </li>
)

export default MenuItem
