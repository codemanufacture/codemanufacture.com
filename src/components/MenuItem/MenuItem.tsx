import * as React from 'react'
import { Link } from 'gatsby'

interface MenuItemProps {
  name: string
  pagePath: string
}

const MenuItem: React.SFC<MenuItemProps> = ({ pagePath, name }) => (
  <li>
    <Link to={pagePath} title={name}>
      {name}
    </Link>
  </li>
)

export default MenuItem
