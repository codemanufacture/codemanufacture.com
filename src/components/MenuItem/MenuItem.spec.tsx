import * as React from 'react'
import renderer from 'react-test-renderer'
import MenuItem from './MenuItem'

describe('MenuItem component', () =>
  it('renders correctly', () => {
    const props = {
      name: 'abc',
      pagePath: 'qwe',
    }

    const tree = renderer.create(<MenuItem {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
