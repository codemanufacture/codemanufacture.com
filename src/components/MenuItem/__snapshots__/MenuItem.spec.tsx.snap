// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component renders correctly 1`] = `
<li>
  <a
    href="/qwe"
    onClick={[Function]}
    onMouseEnter={[Function]}
    title="abc"
  >
    abc
  </a>
</li>
`;
